test/unit/context_adapter_test.js

```diff
@@ -0,0 +1,2079 @@
+/*
+ * Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of context-adapter
+ *
+ * context-adapter is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * context-adapter is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with context-adapter.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[german.torodelvalle@telefonica.com]
+ */
+
+(function(){
+  'use strict';
+
+  var contextAdapter = require('../../lib/context_adapter');
+  var caConfig = require('../../lib/context_adapter_configuration');
+  var caHelper = require('../../lib/context_adapter_helper');
+  var hapi = require('hapi');
+  var request = require('request');
+  var nock = require('nock');
+
+  /**
+   * Returns a valid and well-formed or invalid and not well-formed
+   *  updateContext request depending on the options passed
+   * @param {object} options Object including the properties which should
+   *  be excluded from the final returned payload. For example, if options is:
+   *  {
+   *    contextElements: {
+   *      id: false,
+   *      isPattern: false,
+   *      attributes: [
+   *        caConfig.BUTTON_ENTITY.CA_EXTERNAL_ID_ATTR_NAME,
+   *        caConfig.BUTTON_ENTITY.CA_SERVICE_ID_ATTR_NAME
+   *      ]
+   *    },
+   *    updateAction: false
+   *  }
+   *  the returned payload will not include an id, isPattern properties or
+   *  the specified attributes in the contextElements entry, or an updateAction
+   *  property.
+   * @return {object} The updateContext request payload
+   */
+  function getUpdateContextPayload(options) {
```

Comment:
The following functions seem to be utility functions. Please, extract them to a testUtils module, so the tests can be read more easily.